c++/src/kj/async-inl.h

```diff
@@ -134,6 +134,11 @@ class TraceBuilder {
   void** limit;
 };
 
+struct alignas(void*) PromiseArena {
+  // Space in which a chain of promises may be allocated. See PromiseDisposer.
+  byte bytes[1024];
```

Comment:
I guess if certain applications have deeper promise chains or intrinsically bigger promise nodes, there may be some benefit to having a dynamic promise chain that allocates a new arena twice as large, each time it runs out.  Although the presence of deep promise chains suggests the application might have other problems.

It occurs to me that if the promises in the chain are particularly lightweight, a fair proportion of the arena will be devoted to storing its own address, which could alternately be encoded as a small offset from the promise object itself.  ...I guess that fact might be useful in an alternate formulation of promises, but presumably not worth sacrificing simplicity or natural C++ implementation in this one.

Reply:
I think a bigger potential use for dynamic-sized arenas is that we could maybe be smarter about guessing how big an arena is needed in any particular use case. However, I haven't thought of any clever heuristics for that yet. I think for now having a static size probably helps by reducing code footprint a bit.